docs/serializer.md

```diff
@@ -0,0 +1,171 @@
+* `create_bone_indices_for_single_target_buffer_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_bone_indices_for_single_target_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_indices_for_single_target_buffer_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_2d_texture_coordinates_cached(gvk::serializer& aSerializer, ...)`
+* `create_2d_texture_coordinates_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_2d_texture_coordinates_flipped_cached(gvk::serializer& aSerializer, ...)`
+* `create_2d_texture_coordinates_flipped_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_3d_texture_coordinates_cached(gvk::serializer& aSerializer, ...)`
+* `create_3d_texture_coordinates_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `convert_for_gpu_usage_cached(gvk::serializer& aSerializer, ...)`
+
+# How to use
+The serializer is available via `gvk::serializer` class and can be used for, i.e. improving load
+times of huge orca scenes. It is initialized either by a `gvk::serializer::serialize` or a
+`gvk::serializer::deserialize` object, where both take a path to a **cache** file.
+`gvk::serializer::serialize` creates and writes to the **cache** file and
+`gvk::serializer::deserialize` reads from the **cache** file at the given path. A helper function
+`gvk::does_cache_file_exist(const std::string_view)` can be used to determine which initialization
+method shall be used.
+```
+const std::string cacheFilePath(pathToOrcaScene + ".cache");
+
+auto serializer = gvk::does_cache_file_exist(cacheFilePath) ?
+	gvk::serializer(gvk::serializer::deserialize(cacheFilePath)) :
+	gvk::serializer(gvk::serializer::serialize(cacheFilePath));
+```
+After initialization the current **mode** of the serializer can be retrived by calling `mode()` on
+the serializer, which either returns **serializer::mode::serialize** or
+**serializer::mode::deserialize**. To serialize or deserialize a type, the `archive(Type&&)`
+function template is called on the serializer object. This function serializes the given object to
+the **cache** file if the mode of the serialize is **serialize** or deserializes the object from the
+**cache** file into the given object if the mode is **deserialize**. The following code shows how a
+typical serializer usage can look like when used during orca scene loading.
+```
+std::vector<glm::vec3> positions;
+if (serializer.mode() == gvk::serializer::mode::serialize) {
+	get_positions(orca_scene, positions);
+}
+serializer.archive(positions);
+
+size_t numPositions = (serializer.mode() == gvk::serializer::mode::serialize) ? positions.size() : 0;
+serializer.archive(numPositions);
+```
+
+The serializer features spezialised functions for serializing and deserializing a block of memory
+and host visible buffers. Both can be used separately but are especially usefull when used in
+conjunction to, i.e. serialize/deserialize image data. Image data loaded from a file can be
+serialized using `archive_memory`
```

Comment:
```suggestion
serialized using `gvk::archive_memory`:
```
We should include the namespace to avoid being ambiguous. (Also in other places wherever it applies.)